tests/test.py

```diff
@@ -339,6 +329,64 @@ class MockResponse(object):
             cas_client.acquire_auth_token_ticket()
         m.assert_called_with(
             'https://dummy.url/cas/api/auth_token_tickets',
+            data=None,
             headers={'baz': 'quux'},
             verify=False,
             )
+
+    def _get_test_api_url(self):
+        cas_client = CASClient('https://dummy.url')
+        api_resource = 'do_something_useful'
+        auth_token_ticket = 'ATT-1234'
+        authenticator = 'my_company_ldap'
+        with open(self.private_key_filepath, 'r') as file_pointer:
+            private_key = file_pointer.read()
+        service_url = 'https://example.com'
+        kwargs = {
+            'and': 'another_thing',
+            'you': 'should_know',
+        }
+        return cas_client.get_api_url(
+            api_resource=api_resource,
+            auth_token_ticket=auth_token_ticket,
+            authenticator=authenticator,
+            private_key=private_key,
+            service_url=service_url,
+            **kwargs
+        )
+
+    def test_perform_api_request(self):
+        class MockResponse(object):
+            text = '{"something": "useful"}'
+            success = True
+        cas_client = CASClient('https://dummy.url')
+        url = self._get_test_api_url()
+        service_ticket = 'ST-14600760351898-0B3lSFt2jOWSbgQ377B4CtbD9uq0MXR9kG23vAuH'
+
+        assert not cas_client.headers
+        with mock.patch('cas_client.CASClient._perform_post') as m:
+            m.return_value = MockResponse()
+            response = cas_client.perform_api_request(
+                url,
+                method='POST',
+                body={
+                    'st': service_ticket
```

Comment:
/meganit I find the style [as above on R333](https://github.com/discogs/python-cas-client/pull/12/files#diff-b131fe2646b3e4980d06f90692dc2401R333) preferable here, but if you're just e.g. running `black` on these, nbd.